internal/engine/common.go

```diff
@@ -30,9 +31,19 @@ func compareCommandOutputs(
 	actualOutput string,
 	expectedOutput string,
 	expectedSimilarity float64,
+	expectedRegex string,
 	expectedOutputLanguage string,
 ) error {
-	if strings.ToLower(expectedOutputLanguage) == "json" {
+	if expectedRegex != "" {
+		re, err := regexp.Compile(expectedRegex)
+		if err != nil {
+			return fmt.Errorf(ui.ErrorMessageStyle.Render(fmt.Sprintf("Cannot compile the following regex: %q.", expectedRegex)))
+		}
+
+		if !re.MatchString(actualOutput) {
+			return fmt.Errorf(ui.ErrorMessageStyle.Render(fmt.Sprintf("Expected output does not match: %q.", expectedRegex)))
+		}
+	} else if strings.ToLower(expectedOutputLanguage) == "json" {
```

Comment:
NIT: Instead of chaining these if statements together, I think we should just `return nil` if the expected Regex matches okay. It does add an extra return statement, but I think that the expectation check for regex is independent of the type of output that is presented and should be logically structured as so.